kmir/src/kmir/convert_json/convert.py

```diff
@@ -98,12 +110,33 @@ def constant_kind_from_dict(js: str | Mapping[str, object]) -> KApply:
         _raise_conversion_error('')
 
 
+def constant_kind_from_func_map(n: int) -> KApply:
+    # This will throw KeyError exceptipn if the key (type id) is not in the map.
+    func = func_map[n]
+    match func:
+        case {'NormalSym': str(s)}:
+            hash_object = hashlib.sha256(s.encode('utf-8'))
+            hash_value = int.from_bytes(hash_object.digest(), 'big')
+            return KApply('constantKindFnDef', (KToken(str(hash_value), KSort('Int'))))
+        case {'IntrinsicSym': str(s)}:
+            return KApply('constantKindIntrinsic', (string_from_dict(s)))
+        case {'NoOpSym': str('')}:
+            return KApply('constantKindNoOp', ())
+        case _:
+            _raise_conversion_error('')
+
+
 def mirconstid_from_dict(n: int) -> KApply:
     return KApply('mirConstId(_)_TYPES_MirConstId_Int', (KToken(str(n), KSort('Int'))))
 
 
 def mirconst_from_dict(js: Mapping[str, object]) -> KApply:
     match js:
+        case {'kind': 'ZeroSized', 'ty': int(ty), 'id': int(id_)}:
```

Comment:
So, it is possible that there are Zero Sized constants which do not refer to function types --- in that case, we want to leave those constants alone. We can check for such constants by examining whether their type id is present in our function map.

Reply:
I didn't know that, thanks for point it out! Addressed.